### DISCUSSION TITLE
It seems trim & quote doesn't work as expect

### DISCUSSION THREAD

**Discussion by quangson91** [BurntSushi/rust-csv#327] [Q&A] [answered]
Thank you for build great crate for read/write csv content.

-------

I have a simple csv content look like:

```
"Index", "Year", "Age", "Name", "Movie"
 1, 1928, 44, "Emil Jannings", "The Last Command, The Way of All Flesh"
```

I tried to read these content with following config:

```
csv::ReaderBuilder::new()
        .has_headers(false)
        .delimiter(b',')
        .trim(Trim::All)
        .quote(b'"')
```

But the result looks wrong.
The data in the first row:

```
[0] = `Index`    <---- OK
[1] = ` "Year"`  <----- Not OK; Expect: `Year` without quote
[2] = ` "Age"`  <------ Not OK; Expect: `Age` without quote
[3] = ` "Name"` <------ Not OK; Expect: `Name` without quote
[4] = ` "Movie"` <------ Not OK; Expect: `Movie` without quote
```

For the seconds csv row, it looks more weird. I have to add flexbile to true. Then it parse to 6 fields.

```
 1, 
1928, 
44, 
"Emil Jannings", 
"The Last Command,
The Way of All Flesh"
```

Am I missing any config?
Please help.

Thanks!

**Comment by BurntSushi** [comment, marked as answer]
Yes, what you're missing is that your CSV data is invalid and the csv crate is trying to salvage it. This has nothing to do with the trim or quote options. You can't put spaces between fields in the csv data.

You need to either fix your data or write bespoke code to fix it after parsing.

**Reply by quangson91** [reply]
Ah, I got it.
I also try to open that csv file on Excel or Google Sheet, and I get the same result. 

Thank you  for the explanation!
